### PR TITLE
Better documentation

### PR DESCRIPTION
**Pull request by skellet0r**

### CODE REVIEW

**Review by charles-cooper — APPROVED** [review]
this looks great. nice template and good docs for register/deregister precompile + patch_opcode. lmk when ready and i can merge.

i think the readme should be kept as is as a kind of "quick start" reference but the rtd docs can house the full API reference.

File: docs/source/api.rst [hunk 1]
```diff
@@ -220,9 +220,26 @@ High-Level Functionality
...
+
+.. class:: Env
+
+    .. attribute:: chain
+        :type: eth.abc.ChainAPI
+
+    .. attribute:: eoa
+        :type: str
+
+        The account to use as ``tx.origin`` when performing state mutating contract operations.
```
Comment (charles-cooper) [inline docs/source/api.rst:236]: `env.eoa` sets both `tx.origin` and `msg.sender`

**Comment by skellet0r** [comment]
@charles-cooper just about ready, just a matter of organization left really.

**Comment by charles-cooper** [comment]
thank you!!! :fire:

**Pull request merged**